plugins/modules/ndo_virtual_port_channel_interface.py

```diff
@@ -0,0 +1,383 @@
+  cisco.mso.ndo_virtual_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    description: My Ansible Port Channel
+    virtual_port_channel_interface: ansible_virtual_port_channel_interface
+    node_1: 101
+    node_2: 102
+    interfaces_node_1:
+      - 1/1
+      - 1/10-11
+    interfaces_node_2:
+      - 1/2
+    interface_policy_group:
+      name: ansible_policy_group
+      template: ansible_fabric_policy_template
+    interface_descriptions:
+      - node: 101
+        interface_id: 1/1
+        description: My first Ansible Interface for first node
+      - node: 101
+        interface_id: 1/10
+        description: My second Ansible Interface for first node
+      - node: 101
+        interface_id: 1/11
+        description: My third Ansible Interface for first node
+      - node: 102
+        interface_id: 1/2
+        description: My first Ansible Interface for second node
+    state: present
+
+- name: Query an Virtual Port Channel Interface with template_name
+  cisco.mso.ndo_virtual_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    virtual_port_channel_interface: ansible_virtual_port_channel_interface
+    state: query
+  register: query_one
+
+- name: Query all Virtual Port Channel Interfaces in the template
+  cisco.mso.ndo_virtual_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete an Virtual Port Channel Interface
+  cisco.mso.ndo_virtual_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    virtual_port_channel_interface: ansible_virtual_port_channel_interface
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        virtual_port_channel_interface=dict(type="str", aliases=["name", "virtual_port_channel", "vpc"]),
+        virtual_port_channel_interface_uuid=dict(type="str", aliases=["uuid", "virtual_port_channel_uuid", "vpc_uuid"]),
+        description=dict(type="str"),
+        node_1=dict(type="str"),
+        node_2=dict(type="str"),
+        interfaces_node_1=dict(type="list", elements="str", aliases=["interfaces_1", "members_1"]),
+        interfaces_node_2=dict(type="list", elements="str", aliases=["interfaces_2", "members_2"]),
```

Comment:
should these not be required? how would you create a new vpc without node/interface config? for the update scenario I can understand it can be not provided and that the update is then ignored. else we expect a removal operation is also possible?

Reply:
The API's errors out if they are missing during creation